### ISSUE TITLE
Is this ready for clap-rs org?

### ISSUE THREAD

**Issue by epage** [clap-rs/clap-color-flag#2]
I would prefer experiments happen within our personal orgs before deciding that we support them officially and move them to a proper org.

**Comment by pksunkara** [comment]
I guess the question is what are we looking for?

I didn't add anything extra. I basically copied the work from `colorchoice_clap` crate and removed the duplication as discussed.

The differences in behaviour is described at https://rust-lang.zulipchat.com/#narrow/stream/220302-wg-cli/topic/concolor.20renaming/near/350282694.

It also has a stable 1.0.0 release because of `colorchoice_clap`. So, I am not sure why we are considering this experimental.

**Comment by epage** [comment]
You unilaterally decided that this is the design and name for the crate and put it under clap-rs without consolutation, putting us on the hook for maintaining it.  I do not appreciate that and have asked kbknapp to transfer this crate to your account.

**Comment by pksunkara** [comment]
Moved out.

So, the question now is, is this ready to be moved to `clap-rs` org? I consider it ready. Are there any questions from your side on the naming or the design of the API?

**Comment by pksunkara** [comment]
This uses 2 dependencies: `clap` (with `color` flag) and `colorchoice`. The reasons why I preferred it as `clap-color-flag` instead of `colorchoice-clap` is:

1. We already have `clap-verbosity-flag` as a pattern.
2. `clap` is built on `anstream` and `anstyle` and `colorchoice`. Thus, having this crate in the `anstyle` workspace when it depends on a crate (`clap`) outside the workspace that depends on crates from the workspace appears too much entanglement of dependencies in the perspective of separation of concerns regarding projects.

**Comment by epage** [comment]
From a users perspective, this is tied to the anstream/anstyle ecosystem, not clap.  I could see us also adding support for other CLI parsers.

**Comment by pksunkara** [comment]
> this is tied to the anstream/anstyle ecosystem

The `clap-verbosity-flag` is also tied to the `log` ecosystem, but we have it in clap because it has logic for handling the flags. From a consistency standpoint, the same argument here is about having extra logic for handling the flag.

> I could see us also adding support for other CLI parsers.

Yes, but they need to be in their own parsers, otherwise, the development will become messy due to the intertwining I mentioned above.

This is a similar scenario to what happened with the diesel migration system. We initially had all the migration helpers for other libraries (`barrel` etc..) in `diesel` itself. But then, it started giving us issues when developing the diesel migration system.

**Comment by epage** [comment]
> Yes, but they need to be in their own parsers, otherwise, the development will become messy due to the intertwining I mentioned above.

I disagree about this being "messy".  If you can point to a concrete problem we would run into, I'm open to it.   I see no practical difference in whether the crates we depend on also depend on items within the same workspace.  It inevitably happens.  I regularly run into it with `snapbox`, `criterion`, etc.
